src/meta/client/MetaClient.cpp

```diff
@@ -1323,17 +1318,31 @@ StatusOr<SchemaVer> MetaClient::getNewestEdgeVerFromCache(const GraphSpaceID& sp
     return it->second;
 }
 
-
 folly::Future<StatusOr<bool>> MetaClient::heartbeat() {
     if (clusterId_.load() == 0) {
         clusterId_ = ClusterIdMan::getClusterIdFromFile(FLAGS_cluster_id_path);
     }
     cpp2::HBReq req;
+    req.set_in_storaged(inStoraged_);
     nebula::cpp2::HostAddr thriftHost;
     thriftHost.set_ip(localHost_.first);
     thriftHost.set_port(localHost_.second);
     req.set_host(std::move(thriftHost));
     req.set_cluster_id(clusterId_.load());
+
+    if (inStoraged_ && listener_ != nullptr) {
+        std::unordered_map<GraphSpaceID, std::vector<PartitionID>> leaderIds;
+        listener_->fetchLeaderInfo(leaderIds);
+        if (leaderIds_ != leaderIds) {
```

Comment:
I've been thinking about this, if we want to use the leader info in heartbeat later, we'd better include it in every request. Otherwise, we can't judge the difference from stale old data and new data. 

But for this PR, it is not important, let me take care of it in another PR.

Reply:
OK, I will make thie report the leader info in every HB request.

Reply:
done!